app/src/main/kotlin/com/softteco/template/data/di/DataStoreModule.kt

```diff
@@ -24,49 +26,60 @@ import kotlinx.coroutines.SupervisorJob
 import javax.inject.Named
 import javax.inject.Singleton
 
-private const val USER_PREFERENCES = "user_data_preferences"
-private const val USER_ENCRYPTED = "user_encrypted_data"
 private const val APP_THEME_PREFERENCES = "theme_mode"
+private const val USER_PROFILE_ENCRYPTED = "user_profile_encrypted_data"
+private const val AUTH_TOKEN_ENCRYPTED = "auth_token_encrypted"
 
 @Module
 @InstallIn(SingletonComponent::class)
 object DataStoreModule {
     @Provides
     @Singleton
-    fun providePreferencesDataStore(@ApplicationContext appContext: Context): DataStore<Preferences> {
-        return PreferenceDataStoreFactory.create(
+    @Named("authToken")
```

Comment:
Adding `@Named("authToken")` looks redundant. 
It could be renamed and as it has AuthTokenDto type, it will automatically use the current data store. 
Just could be renamed:
```

    @Provides
    @Singleton
    fun provideAuthTokenEncryptedDataStore(@ApplicationContext appContext: Context): DataStore<AuthTokenDto> {
        return DataStoreFactory.create(
            corruptionHandler = ReplaceFileCorruptionHandler(
                produceNewData = { AuthTokenDto("") }
            ),
            scope = CoroutineScope(Dispatchers.IO + SupervisorJob()),
            produceFile = { appContext.dataStoreFile(AUTH_TOKEN_ENCRYPTED) },
            serializer = AuthTokenSerializer(CryptoManager())
        )
    }
```